FizzBuzz 1/FizzBuzz/Program.cs

```diff
@@ -1,122 +1,106 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Security;
 
 namespace FizzBuzz
 {
     class Program
     {
-        static Boolean ReadYN(string message)
+        static Boolean ReadYN(string prompt)
         {
-            char answer = 'a';
-            var valid = new List<char>();
-            valid.Add('y');
-            valid.Add('n');
-            while (!valid.Contains(answer))
+            string answer = "";
+            do
             {
-                Console.Write(message);
+                Console.Write(prompt);
                 try
                 {
-                    answer = Char.Parse(Console.ReadLine());
-                    answer = Char.ToLower(answer);
+                    answer = Console.ReadLine().ToLower();
                 }
                 catch
                 {
                 }
-            }
-            if (answer == 'y')
-                return true;
-            else
-                return false;
+            } while (answer != "y" && answer != "n");
+
+            return answer == "y";
         }
 
-        static void Main(string[] args)
+        static bool[] GetRules()
         {
-            int max = -1;
-            while (max < 1)
-            {
-                Console.Write("Enter a number to go up to: ");
-                try
-                {
-                    max = Int32.Parse(Console.ReadLine());
-                } catch
-                {
-                }
-            }
+            var rules = new bool[6];
+
+            rules[0] = ReadYN("Enable Fizz if mod 3? (y/n): ");
+            rules[1] = ReadYN("Enable Buzz if mod 5? (y/n): ");
+            rules[2] = ReadYN("Enable Bang if mod 7? (y/n): ");
+            rules[3] = ReadYN("Enable Bong if mod 11? (y/n): ");
+            rules[4] = ReadYN("Enable Fezz if mod 13? (y/n): ");
+            rules[5] = ReadYN("Enable reverse if mod 17? (y/n): ");
```

Comment:
I think you can combine the array creating and assignment using something like:
```
var rules = new bool[]
{
  ReadYN("..."),
  ReadYN("..."),
  ...
};
```
Then you can even combine it with the return statement 🙂